crates/sui-framework/packages/move-stdlib/sources/ascii.move

```diff
@@ -67,18 +67,48 @@ module std::ascii {
         true
     }
 
+    /// Push a `Char` to the end of the `string`.
     public fun push_char(string: &mut String, char: Char) {
         string.bytes.push_back(char.byte);
     }
 
+    /// Pop a `Char` from the end of the `string`.
     public fun pop_char(string: &mut String): Char {
         Char { byte: string.bytes.pop_back() }
     }
 
+    /// Returns the length of the `string` in bytes.
     public fun length(string: &String): u64 {
         string.as_bytes().length()
     }
 
+    /// Append the `other` string to the end of `string`.
+    public fun append(string: &mut String, other: String) {
+        string.bytes.append(other.bytes)
+    }
+
+    /// Insert the `other` string at the `at` index of `string`.
+    public fun insert(s: &mut String, at: u64, o: String) {
+        assert!(at <= s.length(), EInvalidIndex);
+        let l = s.length();
+        let mut front = s.sub_string(0, at);
+        let end = s.sub_string(at, l);
+        front.append(o);
+        front.append(end);
+        *s = front;
+    }
+
+    /// Copy the slice of the `string` from `i` to `j` into a new `String`.
+    public fun sub_string(string: &String, mut i: u64, j: u64): String {
+        assert!(i <= j && j <= string.length(), EInvalidIndex);
```

Comment:
maybe split over multiple asserts for better line numbers with `#[error]` in the future

Reply:
I will do once we get to better error handling. I'd like to change the codes as well, so it's better done in a separate PR.